tests/run/test_patma.py

```diff
@@ -0,0 +1,3454 @@
+### COPIED FROM CPython 3.12 alpha (July 2022)
+### Original part after ############
+# cython: language_level=3
+
+# new code
+import cython
+from Cython.TestUtils import py_parse_code
+
+
+if cython.compiled:
+    def compile(code, name, what):
+        assert what == 'exec'
+        py_parse_code(code)
+
+
+def disable(func):
+    pass
+
+
+############## SLIGHTLY MODIFIED ORIGINAL CODE
+import array
+import collections
+import enum
+import inspect
+import sys
+import unittest
+
+if sys.version_info > (3, 10):
+    import dataclasses
+    @dataclasses.dataclass
+    class Point:
+        x: int
+        y: int
+else:
+    # predates dataclasses with match args
+    class Point:
+        __match_args__ = ("x", "y")
+        x: int
+        y: int
+
+# TestCompiler removed - it's very CPython-specific
```

Comment:
I usually do this by putting the whole class in triple quotes, just to keep the diff easy to read.